### PR TITLE
TorrentHaven: Fix date parsing. Resolves #7983

### LINKED ISSUE #7983: [torrentheaven] Exception (torrentheaven): DateTime parsing failed for
Exception (torrentheaven): DateTime parsing failed for "Today 13:57   Uploader: Invisible": System.FormatException: The string '13:57   Uploader: Invisible' was not recognized as a valid DateTime. There is an unknown word starting at index '16'. at System.DateTimeParse.Parse(ReadOnlySpan`1 s, DateTimeFormatInfo dtfi, DateTimeStyles styles) at System.DateTime.Parse(String s) at Jackett.Common.Utils.DateTimeUtil.ParseTimeSpan(String time) in /home/vsts/work/1/s/src/Jackett.Common/Utils/DateTimeUtil.cs:line 291 at Jackett.Common.Utils.DateTimeUtil.FromUnknown(String str, String format) in /home/vsts/work/1/s/src/Jackett.Common/Utils/DateTimeUtil.cs:line 118: Parse error

### PR DESCRIPTION
**Pull request by cadatoiva**
no They added an uploader line into the same element, no good way to remove it without search and remove.

### CODE REVIEW

File: src/Jackett.Common/Indexers/TorrentHeaven.cs [hunk 1]
```diff
@@ -95,24 +78,17 @@ private new ConfigurationDataCaptchaLogin configData
             AddCategoryMapping(71, TorznabCatType.PCMac, "APPLICATIONS/Mac");
         }
 
-        public override async Task<ConfigurationData> GetConfigurationForSetup()
+        private new ConfigurationDataCaptchaLogin configData => (ConfigurationDataCaptchaLogin)base.configData;
+
+        private string IndexUrl => SiteLink + "index.php";
+
+        public override string[] LegacySiteLinks { get; protected set; } =
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/TorrentHeaven.cs:85]: Why are you moving these attributes after the constructor?
Fields should be before constructor => https://stackoverflow.com/a/310967

File: src/Jackett.Common/Indexers/TorrentHeaven.cs [hunk 2]
```diff
@@ -202,12 +198,14 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
                     var catStr = qCatLink.GetAttribute("href").Split('=')[3].Split('#')[0];
                     var link = new Uri(SiteLink + qDlLink.GetAttribute("href"));
                     var sizeStr = qSize.TextContent;
-                    var dateStr = qDateStr.TextContent.Trim().Replace("Heute", "Today").Replace("Gestern", "Yesterday");
-
+                    var dateStr = qDateStr.TextContent;
+                    var split = dateStr.IndexOf("Uploader", StringComparison.OrdinalIgnoreCase);
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/TorrentHeaven.cs:202]: For me is much more clear if you do
```
var dateSplit = dateStr.Split(new[] { "Uploader" }, StringSplitOptions.None);
```
and then you take the parts you need

File: src/Jackett.Common/Indexers/TorrentHeaven.cs (hunk 1 above)
Comment (cadatoiva) [reply src/Jackett.Common/Indexers/TorrentHeaven.cs:85]: The default ReSharper setting moves them that way, so when I run the code cleanup (Ctrl+E, C) this is the result.
image

File: src/Jackett.Common/Indexers/TorrentHeaven.cs (hunk 2 above)
Comment (cadatoiva) [reply src/Jackett.Common/Indexers/TorrentHeaven.cs:202]: `IndexOf(string)` stops at the first one that's found, and `SubString(start, length)`. Split finds every instance and puts them all in an array.

Comment (ngosang) [reply src/Jackett.Common/Indexers/TorrentHeaven.cs:202]: I know your implementation is a bit more performant but I still prefer
`dateStr = dateStr.Split(new[] { "Uploader" }, StringSplitOptions.None).First().Trim()
.Replace("Heute", "Today").Replace("Gestern", "Yesterday");`
When there are no big savings I prefer that it be readable.

**Pull request merged**